packages/compute-baseline/src/browser-compat-data/supportStatements.ts

```diff
@@ -198,3 +278,30 @@ export class RealSupportStatement extends SupportStatement {
     }));
   }
 }
+
+function statementToQualifications(
+  statement: SupportStatement,
+): Qualifications {
+  let qualifications: Qualifications = {};
+  if (statement.data.prefix) {
+    qualifications.prefix = statement.data.prefix;
+  }
+  if (statement.data.alternative_name) {
+    qualifications.alternative_name = statement.data.alternative_name;
+  }
+  if (statement.flags.length) {
+    qualifications.flags = statement.flags;
+  }
+  if (statement.partial_implementation) {
+    qualifications.partial_implementation = statement.partial_implementation;
+  }
+  return qualifications;
+}
+
+function isRangedVersion(s: any): s is string {
```

Comment:
I've only seen this syntax once before, but does `s is string` really do the right thing if this method returns false? Won't TypeScript infer that the argument wasn't a string even though it may have been?

Reply:
Yes, this really does the right thing. Type predicates are (by themselves) one-way: if the function returns `true`, then the type is satisfied. The inverse (`false` → s is not string) is not inferred. [Demo](https://www.typescriptlang.org/play/?#code/DYUwLgBGIM5gXBArgOwNYoPYHcUQLwQBEmmAFgDQRwBOAligOYCeRA3AFCiQqIpIBbAEYgaBCAEYATAGYALJ27UwNRLQaNxRdUzqj2EDhwBmqAMZg6mPHRgBlFRuYAKGIlQYcKAJRqIt5XomCABvDggIiBpwJBo8MGYABxBMY2oCfEJtRyYiCAAyfOoAOgYzYCQAE1hnbKCWIm9OAF8jOjTnBOTUqFhITKydRkbQ8Mj2iE6+jMIfUciFiAB6JYhRGkxVCCwITAA3UWAAQ0SIETBsEBA8IYgjlErtwREaMYjW1o4J51sHepdoHBvN55uMOoD+pltiCwotIis1jQNlsdvtDiczuBLtdAho7g8nsJRG8IB81sAYCBQRFvhCZoEYSSFgidutNpizEckJTenB-DBlHRgMAIAByDxYXCikkfDhAA).

AFAICT, even in general TypeScript doesn't actually track the list of types something _isn't_, just the set of remaining possible types.